### PR TITLE
build/configs: modify esp32 link script's IROM/DROM layout

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
the main modifcation is as below.
1) modify iram0_2_seg length.
2) modify drom0_0_seg offset and length.

and the modification is  originated from esp-idf 
(1) commit id: c230e4280c039b0b709d54e974f63eed358c1c2a
Limit DROM/IROM section lengths correctly

(2) d92c541b1a75d7ced244e73b5c840511f03e348a
 esptool: Optimise generated image size by using RAM-loaded data for padding

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [07538457118998bd7ab8f02d76647f78e75e083a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [07538457118998bd7ab8f02d76647f78e75e083a] - Code Rule Check (C++) OK.

**Comment by xiarihf** [comment]
OK.

File: build/configs/esp32_DevKitC/scripts/esp32.template [hunk 1]
```diff
@@ -47,7 +47,7 @@ MEMORY
 
   /* Even though the segment name is iram, it is actually mapped to flash */
 
-  iram0_2_seg (RX) :     org = 0x400D0018, len = 0x330000
+  iram0_2_seg (RX) :     org = 0x400D0018, len = 0x330000-0x18
```
Comment (sunghan-chang) [inline build/configs/esp32_DevKitC/scripts/esp32.template]: Why? I can't find why you modify like this at anywhere.

**Comment by seinfra** [comment]
Target : [eab2fafc8e1433eb9c2a1006befc4235cba2860c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [eab2fafc8e1433eb9c2a1006befc4235cba2860c] - Code Rule Check (C++) OK.

File: build/configs/esp32_DevKitC/scripts/esp32.template (hunk 1 above)
Comment (ChinaTizenRT) [reply build/configs/esp32_DevKitC/scripts/esp32.template]: okay,  added  comment for modifications

**Pull request merged**